AsyncDisplayKit/Details/ASCollectionViewFlowLayoutInspector.h

```diff
@@ -47,12 +50,37 @@
  */
 - (void)didChangeCollectionViewDelegate:(id<ASCollectionDelegate>)delegate;
 
+/**
+ * Allow the inspector to respond to dataSource changes.
+ *
+ * @discussion A great time to update perform selector caches!
+ */
+- (void)didChangeCollectionViewDataSource:(id<ASCollectionDataSource>)dataSource;
+
 @end
 
+/**
+ * A layout inspector for non-flow layouts that returns a constrained size to let the cells layout itself as
+ * far as possible based on the scrollable direction of the collection view. It throws exceptions for delegate
+ * methods that are related to supplementary node's management.
+ */
+@interface ASCollectionViewDefaultCustomLayoutInspector : NSObject <ASCollectionViewLayoutInspecting>
+
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initWithCollectionView:(ASCollectionView *)collectionView NS_DESIGNATED_INITIALIZER;
+
+@end
+
+/**
+ * A layout inspector implementation specific for the sizing behavior of UICollectionViewFlowLayouts
+ */
 @interface ASCollectionViewFlowLayoutInspector : NSObject <ASCollectionViewLayoutInspecting>
```

Comment:
Sounds good to me!
    
  
  
>   
> On Jun 24, 2016 at 19:42,  <Michael Schneider (mailto:notifications@github.com)>  wrote:
>   
>   
>   
>
> In   AsyncDisplayKit/Details/ASCollectionViewFlowLayoutInspector.h (https://github.com/facebook/AsyncDisplayKit/pull/1755#discussion_r68425747):
>
>   
>   >  @end  >   >  +/**  >  + * A layout inspector for non-flow layouts that returns a constrained size to let the cells layout itself as  >  + * far as possible based on the scrollable direction of the collection view. It throws exceptions for delegate  >  + * methods that are related to supplementary node's management.  >  + */  >  +@interface ASCollectionViewDefaultCustomLayoutInspector : NSObject  <ASCollectionViewLayoutInspecting>   >  +  >  +- (instancetype)init NS_UNAVAILABLE;  >  +- (instancetype)initWithCollectionView:(ASCollectionView *)collectionView NS_DESIGNATED_INITIALIZER;  >  +  >  +@end  >  +  >  +/**  >  + * A layout inspector implementation specific for the sizing behavior of UICollectionViewFlowLayouts  >  + */  >  @interface ASCollectionViewFlowLayoutInspector : NSObject  <ASCollectionViewLayoutInspecting>   
>   
>
> As I propose to go with  ASCollectionViewLayoutInspector  I think we should not rename the class. So it would be  ASCollectionViewLayoutInspector  and  ASCollectionViewFlowLayoutInspector.
>
>   
>
> —
>  You are receiving this because you were mentioned.
>  Reply to this email directly,   view it on GitHub (https://github.com/facebook/AsyncDisplayKit/pull/1755/files/5957bea8594f1419b72a30fd00bb527bd243408b#r68425747), or   mute the thread (https://github.com/notifications/unsubscribe/AAAhLVUamieacFjDoifR3LzpCKdFR8OJks5qPAkPgaJpZM4I3pap).
>
>   
>   
>     
>   
>   
>